storage/innobase/row/row0upd.cc

```diff
@@ -3485,9 +3485,16 @@ void upd_node_t::make_versioned_helper(const trx_t* trx, ulint idx)
 
 	dict_index_t* clust_index = dict_table_get_first_index(table);
 
+	ut_ad(update->n_fields < table->n_cols + table->n_v_cols);
+
+	upd_field_t* new_buf = static_cast<upd_field_t*>(mem_heap_alloc(
+	    update->heap, (update->n_fields + 1) * sizeof(update->fields[0])));
```

Comment:
Do we actually have to copy the update vector? The added `ut_ad()` should already prove that the capacity that was pre-allocated in `row_create_update_node_for_mysql()` is sufficient.